src/dmd/mtype.d

```diff
@@ -7233,6 +7233,7 @@ ScopeRef buildScopeRef(bool returnByRef, StorageClass stc) pure nothrow @nogc @s
     ScopeRef result;
     final switch (stc & (STC.ref_ | STC.scope_ | STC.return_))
     {
+        case STC.return_:
```

Comment:
What's this doing? It's not covered by tests.

Reply:
Actually, it is, which is why I put it in.

Reply:
Codecov says "Added line #L7236 was not covered by tests"

My question is, why is a single `return` equal to `ScopeRef.None`? If anything, I'd say it's `ScopeRef.Scope` since `return int*` and `return scope int*` are the same. IIRC dmd adds `STC.scope_` in that case, so when is there a single `STC.return_`?

Reply:
It's covered by the library. Remove it and the library will fail to compile.

`return` by itself is used for other things, like returning through the first argument.

Reply:
Anyhow, what I'm moving towards with all these PRs is the ability to solve the `ref return scope` problem.

Reply:
> `return` by itself is used for other things, like returning through the first argument.

Can you add a comment about that to the code? Currently the `case STC.return_:` stands out as an oddity.

Reply:
Added a test case for it.